src/include/verbose.h

```diff
@@ -246,6 +289,37 @@ struct __wt_verbose_multi_category {
         }                                                                                         \
     } while (0)
 
+/*
+ * __wt_verbose_level_multi --
+ *     Display a verbose message, given a set of multiple verbose categories. A verbose message will
+ *     be displayed if at least one category in the set satisfies the required verbosity level.
+ */
+#define __wt_verbose_level_multi_id(session, multi_category, verb_level, log_id, fmt, ...)        \
+    do {                                                                                          \
+        uint32_t __v_idx;                                                                         \
+        /*                                                                                        \
+         * multi_category can be a ternary expression that returns one of two structs. If we call \
+         * it 3 times in this macro then we're evaluating that ternary 3 times and could return a \
+         * different value on a second call. Save it into a local variable to make sure we're     \
+         * working with a constant value.                                                         \
+         */                                                                                       \
+        WT_VERBOSE_MULTI_CATEGORY __multi_category = multi_category;                              \
+        for (__v_idx = 0; __v_idx < __multi_category.cnt; __v_idx++) {                            \
+            if (WT_VERBOSE_LEVEL_ISSET(                                                           \
+                  session, __multi_category.categories[__v_idx], verb_level)) {                   \
+                /* Create a message info structure to pass the log ID and category */             \
+                WT_VERBOSE_MESSAGE_INFO verb_message_info = {                                     \
```

Comment:
I would suggest moving this structure assignment into a separate macro that calls the function `__wt_verbose_worker_id`. So that we can avoid duplicating the code where the structure assignment happens.

Reply:
Fixed in 0f136343cc588b81524565f6e764cc7d7311d112